web/src/components/ShareIcon.tsx

```diff
@@ -0,0 +1,20 @@
+import { Capacitor } from '@capacitor/core';
+import { Share, Share2 } from 'lucide-react';
+import { DEFAULT_ICON_SIZE } from 'utils/constants';
+
+export function ShareIcon({
+  showIosIcon = defaultShouldShowIosIcon(),
+  iconSize = DEFAULT_ICON_SIZE,
+}: {
+  showIosIcon?: boolean;
+  iconSize?: number;
+}) {
+  return showIosIcon ? (
+    <Share data-test-id="iosShareIcon" size={iconSize} />
+  ) : (
+    <Share2 data-test-id="defaultShareIcon" size={iconSize} />
+  );
+}
+
+export const defaultShouldShowIosIcon = () =>
+  /Mac|iPad|iPhone|iPod/.test(navigator.userAgent) || Capacitor.getPlatform() === 'ios';
```

Comment:
Don't we test this regex somewhere else as well? would be nice to only have it defined once.

Reply:
Haha, I think that's in my other PR that I haven't merged in just yet...